### DISCUSSION TITLE
I am getting unknown in crashlytics console

### DISCUSSION THREAD

**Discussion by gresi29** [invertase/react-native-firebase#5080] [Q&A]
I want to track stacktrace using @react-native-firebase/crashlytics in my application. for that i followed all setup steps which are defined in [this](https://rnfirebase.io/crashlytics/usage).

### after setup completed, i created error like this,
setWeekData(){
    console.log(year);  //where year is not defined
 }

### and report error and set attribute like this, 
componentDidCatch(error, errorInfo) {
   crashlytics().setAttribute('error details', errorInfo.componentStack);
   crashlytics().recordError(error);
 }

after this i run this code in debug mode in android and ios that time i got proper error in firebase console.
here i attached screenshots of console log
![download (3)](https://user-images.githubusercontent.com/54353829/112634233-58a2c900-8e60-11eb-8361-df914b87cfca.png)
![download (4)](https://user-images.githubusercontent.com/54353829/112634594-d1098a00-8e60-11eb-832c-97d0059b9404.png)

but when this error occurred in release build (in release apk/ipa), i'm getting unknown instead of function name and filename.
In firebase console i got like,
![2021-03-26 (1)](https://user-images.githubusercontent.com/54353829/112634537-ba633300-8e60-11eb-9c88-ef389fb8ee56.png)

![download (1)](https://user-images.githubusercontent.com/54353829/112633368-48d6b500-8e5f-11eb-8c9b-c24932f0f93e.png)

So please tell me, at which point i was going wrong? Hope i will get your response soon.
Thank you.

**Comment by mikehardy** [comment]
There's nothing wrong. This is the whole point of the release mode vs debug mode build types. Release mode is minified and frequently obfuscated, it's basically unreadable but it is much smaller and much higher performance because of the minification (and maybe obfuscation) transformations.

You must generate sourcemaps and save them (or tag your build so you can generate them as needed in the future in a way that corresponds with the code the stack came from). Then you use a stack-trace de-obfuscator that combines your sourcemaps and your obfuscated stack trace, and you have a more usable stack trace.

**Reply by vandit560** [reply]
which npm to used for sourcemaps
